avutil/avutil.go

```diff
@@ -980,6 +989,64 @@ func (f *Frame) PacketDuration() int64 {
 	return int64(C.av_frame_get_pkt_duration(f.CAVFrame))
 }
 
+func (f *Frame) GetSideData() *FrameSideData {
+	cSideData := C.av_frame_get_side_data(f.CAVFrame, C.AV_FRAME_DATA_MOTION_VECTORS)
+	if cSideData == nil {
+		return nil
+	}
+
+	return &FrameSideData{
+		CFrameSideData: (*C.AVFrameSideData)(unsafe.Pointer(cSideData)),
+	}
+}
+
+type FrameSideData struct {
+	CFrameSideData *C.AVFrameSideData
+}
+
+func (sd *FrameSideData) Size() int {
+	return int(C.go_motion_vector_size(sd.CFrameSideData))
+}
+
+func (sd *FrameSideData) MotionVector(n int) *MotionVector {
```

Comment:
Please consider this alternative to extract motion vectors:

```go
func (sd *FrameSideData) MotionVectors() []*MotionVector {
   if sd.Type() != FrameSideDataTypeMotionVectors {
     return nil
   }
   count := int(C.go_motion_vector_size(sd.CFrameSideData))
   if count <= 0 {
     return nil
   }
   mvs := make([]*MotionVector, count)
   for i := 0; i < count; i++ {
     mvs[i] = NewMotionVectorFromC(C.go_motion_vector(sd.CFrameSideData, C.int(i)))
   }
   return mvs
}
```

What do you think?